common-money.lic

```diff
@@ -69,6 +106,56 @@ module DRCM
     check_wealth(hometown_currency(hometown))
   end
 
+  def get_total_wealth
+    # This method captures your current total on-hand wealth
+    # and returns a hash representing the numerical value in
+    # coppers of each currency.    
+
+    # Set up variables to capture the value in coppers of each currency
+    # Set to zero so that, if we have, for example, "No Lirums"
+    # we simply return the initialized value of 0.
+    kronars = 0
+    lirums = 0
+    dokoras = 0
+
+    # Grab the character's wealth, pausing a bit
+    # then grabbing a sufficient number of lines
+    # to ensure we get all the output taking into
+    # account other random scroll text.
+    # Reversing the lines ensures we are processing
+    # the most recent output from 'wealth', in case
+    # reget were to grab output from back-to-back calls.
+    DRC.bput("wealth", "Wealth")
+    pause 0.5
+    wealth_lines = reget(10).map(&:strip).reverse
+
+    # We've reversed the reget array. Now we'll iterate over it and capture
+    # each line after we recognize we've hit the Wealth block.
+    wealth_lines.each do |line|
+      case line
+      when /^Wealth:/i
+        # This is the start of our Wealth lines.
+        # We don't need to parse this line. Break out of loop.
+        break
+      when /\(\d+ copper Kronars\)/i
+        kronars = line.scan(/\((\d+) copper Kronars\)/i).first.first.to_i
+      when /\(\d+ copper Lirums\)/i
+        lirums = line.scan(/\((\d+) copper lirums\)/i).first.first.to_i
+      when /\(\d+ copper Dokoras\)/i
+        dokoras = line.scan(/\((\d+) copper dokoras\)/i).first.first.to_i
```

Comment:
💅 Purely for consistency within the regex patterns, suggest they all follow either capital case or lower case. Note the current code has Kronars, lirums, dokoras; a mix of upper and lower case.

Reply:
Done!